readme.md
# DiorgesBB

[![Latest Version on Packagist][ico-version]][link-packagist]
[![Total Downloads][ico-downloads]][link-downloads]
[![Build Status][ico-travis]][link-travis]
[![StyleCI][ico-styleci]][link-styleci]

Pacote para Laravel, API para registro/verificação de boletos usando o projeto Piloto do BB, compátivel com Laravel 5 ao 8.

## Installation

Via Composer

``` bash
$ composer require diorgesl/diorgesbb
```

## Usage

Arquivo de Configuração
``` bash
php artisan vendor:publish --provider="Diorgesl\DiorgesBB\DiorgesBBServiceProvider"
```

Configurar o arquivo `config/diorgesbb.php` com os dados de acesso da API.

Para registrar um boleto:
```
<?php 
use Diorgesl\DiorgesBB\Boletos;
use Diorgesl\DiorgesBB\Boleto;
use Diorgesl\DiorgesBB\Pagador;

$boletos = new Boletos();

$pagador = new Pagador([
    'numeroRegistro' => 97965940132,
    'nome' => 'CLIENTE TESTE',
    'endereco' => 'AL SATELITE 13',
    'bairro' => 'UNIVERSITARIO',
    'cidade' => 'CORUMBA',
    'cep' => 79304310,
    'uf' => 'MS'
]);

$boleto = new Boleto([
    "codigoModalidade" => 1,
    "dataEmissao" => "29.09.2020",
    "dataVencimento" => "30.10.2020",
    "valorOriginal" => 109.90,
    "codigoAceite" => "N",
    "codigoTipoTitulo" => 10,
    "descricaoTipoTitulo" => "Duplicata Mercantil",
    "indicadorPermissaoRecebimentoParcial" => "N",
    "numeroTituloBeneficiario" => "123456",
    "numeroTituloCliente" => 43832319,
    "pagador" => $pagador,
]);

$ret = $boletos->registrar($boleto);

var_dump($ret);
```

Resposta do banco:
```
{
   "numero":"00031285570043832319",
   "numeroCarteira":17,
   "numeroVariacaoCarteira":35,
   "codigoCliente":704950857,
   "linhaDigitavel":"00190000090312855700043832319172884240000010990",
   "codigoBarraNumerico":"00198842400000109900000003128557004383231917",
   "numeroContratoCobranca":19581316,
   "beneficiario":{
      "agencia":452,
      "contaCorrente":123873,
      "tipoEndereco":0,
      "logradouro":"Cliente nao localizado ou sem enderecos validos.",
      "bairro":"",
      "cidade":"",
      "codigoCidade":0,
      "uf":"",
      "cep":0,
      "indicadorComprovacao":""
   },
   "quantidadeOcorrenciasNegativacao":"0",
   "listaOcorrenciasNegativacao":[
      
   ]
}
```

Outras funções:

Dar baixa em um boleto:
```
?php 
use Diorgesl\DiorgesBB\Boletos;

$boletos = new Boletos();

$boletos->baixar(12345);
```

Resposta do banco:
```
{
  "numeroContratoCobranca": "19581316"
  "dataBaixa": "01.10.2020"
  "horarioBaixa": "15:04:33"
}
```
---------
Listar boletos:
```
?php 
use Diorgesl\DiorgesBB\Boletos;

$boletos = new Boletos();

// Retornar todos os boletos baixados/liquidados no periodo do movimento
$boletos = $boletos->boletos([
    "indicadorSituacao" => "B", // A = em aberto (Padrão), B = Baixados/Protestados/Liquidados
    "dataInicioMovimento" => "01.10.2020",
    "dataFimMovimento" => "20.10.2020", 
]);

// Retornar todos os boletos de um CPF/CNPJ
// Todos os boletos em aberto do CPF
$boletos = $boletos->boletos([
    //"indicadorSituacao" => "A", // A = em aberto (Padrão), B = Baixados/Protestados/Liquidados
    "cpfPagador" => 979659401,
    "digitoCPFPagador" => 32,
]);

//Todos os boletos em aberto do CNPJ 
$boletos = $boletos->boletos([
    "cnpjPagador" => 196152360001,
    "digitoCNPJPagador" => 27,
]);
```
---------
Detalhar um Boleto:
```
?php 
use Diorgesl\DiorgesBB\Boletos;

$boletos = new Boletos();
// Busca o boleto com ID 123456
$boleto = $boletos->boleto(123456);
var_dump($boleto);
```

Resposta do banco:
```
{
   "numeroContratoCobranca":"19581316",
   "codigoEstadoTituloCobranca":1,
   "codigoTipoTituloCobranca":10,
   "codigoModalidadeTitulo":1,
   "codigoAceiteTituloCobranca":"N",
   "codigoPrefixoDependenciaCobrador":14,
   "codigoIndicadorEconomico":9,
   "numeroTituloCedenteCobranca":"123456",
   "dataEmissaoTituloCobranca":"29.09.2020",
   "dataRegistroTituloCobranca":"30.09.2020",
   "dataVencimentoTituloCobranca":"30.10.2020",
   "valorOriginalTituloCobranca":109.9,
   "valorAtualTituloCobranca":109.9,
   "valorPagamentoParcialTitulo":0,
   "valorAbatimentoTituloCobranca":0,
   "percentualImpostoSobreOprFinanceirasTituloCobranca":0,
   "valorImpostoSobreOprFinanceirasTituloCobranca":0,
   "valorMoedaTituloCobranca":0,
   "quantidadeParcelaTituloCobranca":0,
   "dataBaixaAutomaticoTitulo":"30.10.2021",
   "textoCampoUtilizacaoCedente":"",
   "indicadorCobrancaPartilhadoTitulo":"N",
   "valorMoedaAbatimentoTitulo":0,
   "dataProtestoTituloCobranca":"",
   "numeroCarteiraCobranca":17,
   "numeroVariacaoCarteiraCobranca":35,
   "quantidadeDiaProtesto":0,
   "quantidadeDiaPrazoLimiteRecebimento":360,
   "dataLimiteRecebimentoTitulo":"25.10.2021",
   "indicadorPermissaoRecebimentoParcial":"N",
   "textoCodigoBarrasTituloCobranca":"00197842400000109900000003128557000000123456",
   "codigoOcorrenciaCartorio":0,
   "indicadorDebitoCreditoTitulo":0,
   "valorImpostoSobreOprFinanceirasRecebidoTitulo":0,
   "valorAbatimentoTotal":0,
   "valorCreditoCedente":0,
   "codigoTipoLiquidacao":0,
   "dataCreditoLiquidacao":"",
   "dataRecebimentoTitulo":"",
   "codigoPrefixoDependenciaRecebedor":0,
   "codigoNaturezaRecebimento":0,
   "codigoResponsavelAtualizacao":"",
   "codigoTipoBaixaTitulo":0,
   "valorReajuste":0,
   "valorOutroRecebido":0,
   "codigoIndicadorEconomicoUtilizadoInadimplencia":0,
   "sacado":{
      "codigoTipoInscricaoSacado":1,
      "numeroInscricaoSacadoCobranca":97965940132,
      "nomeSacadoCobranca":"CLIENTE TESTE",
      "textoEnderecoSacadoCobranca":"RUA TESTE",
      "nomeBairroSacadoCobranca":"TESTE",
      "nomeMunicipioSacadoCobranca":"BRASILIA",
      "siglaUnidadeFederacaoSacadoCobranca":"DF",
      "numeroCepSacadoCobranca":79000000,
      "valorPagoSacado":0,
      "numeroIdentidadeSacadoTituloCobranca":""
   },
   "sacador":{
      "codigoTipoInscricaoSacador":1,
      "numeroInscricaoSacadorAvalista":97965940132,
      "nomeSacadorAvalistaTitulo":"CLIENTE TESTE"
   },
   "multa":{
      "percentualMultaTitulo":0,
      "valorMultaTituloCobranca":0,
      "dataMultaTitulo":"",
      "valorMultaRecebido":0
   },
   "desconto":{
      "percentualDescontoTitulo":0,
      "dataDescontoTitulo":"",
      "valorDescontoTitulo":0,
      "codigoDescontoTitulo":0,
      "valorDescontoUtilizado":0,
      "segundoDesconto":{
         "percentualSegundoDescontoTitulo":0,
         "dataSegundoDescontoTitulo":"",
         "valorSegundoDescontoTitulo":0,
         "codigoSegundoDescontoTitulo":0
      },
      "terceiroDesconto":{
         "percentualTerceiroDescontoTitulo":0,
         "dataTerceiroDescontoTitulo":"",
         "valorTerceiroDescontoTitulo":0,
         "codigoTerceiroDescontoTitulo":0
      }
   },
   "juroMora":{
      "codigoTipoJuroMora":0,
      "percentualJuroMoraTitulo":0,
      "valorJuroMoraTitulo":0,
      "dataJuroMoraTitulo":"",
      "valorJuroMoraRecebido":0
   }
}
```


## Change log

Please see the [changelog](changelog.md) for more information on what has changed recently.

## Testing

``` bash
$ composer test
```

## Contributing

Please see [contributing.md](contributing.md) for details and a todolist.

## Security

If you discover any security related issues, please email diorges@gis.net.br instead of using the issue tracker.

## Credits

- [Diorges Rocha][link-author]
- [Eduardo Kum (Laravel Boleto)][link-eduardo]
- [All Contributors][link-contributors]

## License

MIT. Please see the [license file](license.md) for more information.

[ico-version]: https://img.shields.io/packagist/v/diorgesl/diorgesbb.svg?style=flat-square
[ico-downloads]: https://img.shields.io/packagist/dt/diorgesl/diorgesbb.svg?style=flat-square
[ico-travis]: https://img.shields.io/travis/diorgesl/diorgesbb/master.svg?style=flat-square
[ico-styleci]: https://github.styleci.io/repos/299090003/shield

[link-packagist]: https://packagist.org/packages/diorgesl/diorgesbb
[link-downloads]: https://packagist.org/packages/diorgesl/diorgesbb
[link-travis]: https://travis-ci.org/diorgesl/diorgesbb
[link-styleci]: https://github.styleci.io/repos/299090003
[link-author]: https://github.com/diorgesl
[link-eduardo]: https://github.com/eduardokum/laravel-boleto
[link-contributors]: ../../contributors


changelog.md
# Changelog

All notable changes to `DiorgesBB` will be documented in this file.

## Version 1.0

### Added
- Everything


src/Boleto.php
<?php

namespace Diorgesl\DiorgesBB;

class Boleto implements \JsonSerializable
{
    /*
     * Identificador determinado pelo sistema Cobrança para controlar a emissão
     * de boletos, liquidação, crédito de valores ao Beneficiário e intercâmbio
     * de dados com o cliente. Campo Obrigatório.
     *
     * @var float
     */
    protected $numeroConvenio = 0;

    /*
     * Características do serviço de boleto bancário e como ele deve ser tratado
     * pelo banco. Campo Obrigatório.
     *
     * @var float
     */
    protected $numeroCarteira = 0;

    /*
     * Código do tipo de portfolio do boleto. Todos os portfólios devem ter um
     * código específico. Campo Obrigatório.
     *
     * @var float
     */
    protected $numeroVariacaoCarteira = 0;

    /*
     * Identifica a característica dos boletos dentro das modalidades de cobrança
     * existentes no banco. Campo Obrigatório. Domínio: 01 - SIMPLES; 04 - VINCULADA
     *
     * @var float
     */

    protected $codigoModalidade = 0;

    /*
     * Data de emissão do boleto (formato "dd.mm.aaaaa"). Campo Obrigatório.
     *
     * @var string
     */
    protected $dataEmissao;

    /*
     * Data de vencimento do boleto (formato "dd.mm.aaaaa"), >= dataEmissao.
     * Campo Obrigatório.
     *
     * @var string
     */
    protected $dataVencimento;

    /*
     * Valor de cobrança > 0.00, emitido em Real (formato decimal separado por ".").
     * Valor do boleto no registro. Deve ser maior que a soma dos campos
     * “VALOR DO DESCONTO DO TÍTULO” e “VALOR DO ABATIMENTO DO TÍTULO”, se informados.
     * Informação não passível de alteração após a criação. No caso de emissão com valor
     * equivocado, sugerimos cancelar e emitir novo boleto. Campo Obrigatório.
     *
     * @var float
     */
    protected $valorOriginal = 0.00;

    /*
     * Valor de dedução do boleto >= 0.00 (formato decimal separado por ".").
     *
     * @var float
     */
    protected $valorAbatimento = 0.00;

    /*
     * Quantos dias após a data de emissão do boleto para iniciar o processo
     * de cobrança através de protesto. (valor inteiro >= 0).
     *
     * @var float
     */
    protected $quantidadeDiasProtesto = 0;

    /*
     * Indicador de que o boleto pode ou não ser recebido após o vencimento.
     * Campo não obrigatório Se não informado, será assumido a informação de
     * limite de recebimento que está definida no convênio.
     *
     * @var string
     */
    protected $indicadorNumeroDiasLimiteRecebimento; // S para SIM ou N para NÃO

    /*
     * Número de dias limite para recebimento. Informar valor inteiro > 0 se
     * o campo indicadorNumeroDiasLimiteRecebimento = "S".
     *
     * @var float
     */
    protected $numeroDiasLimiteRecebimento = 0;

    /*
     * Código para identificar se o boleto de cobrança foi aceito (reconhecimento
     * da dívida pelo Pagador). Campo Obrigatório.
     * Domínios: A - ACEITE N - NAO ACEITE
     */
    protected $codigoAceite;

    /*
     * Código para identificar o tipo de boleto de cobrança. Campo Obrigatório.
     * Domínios:
     * 1- CHEQUE
     * 2- DUPLICATA MERCANTIL
     * 3- DUPLICATA MTIL POR INDICACAO
     * 4- DUPLICATA DE SERVICO
     * 5- DUPLICATA DE SRVC P/INDICACAO
     * 6- DUPLICATA RURAL
     * 7- LETRA DE CAMBIO
     * 8- NOTA DE CREDITO COMERCIAL
     * 9- NOTA DE CREDITO A EXPORTACAO
     * 10- NOTA DE CREDITO INDULTRIAL
     * 11- NOTA DE CREDITO RURAL
     * 12- NOTA PROMISSORIA
     * 13- NOTA PROMISSORIA RURAL
     * 14- TRIPLICATA MERCANTIL
     * 15- TRIPLICATA DE SERVICO
     * 16- NOTA DE SEGURO
     * 17- RECIBO
     * 18- FATURA
     * 19- NOTA DE DEBITO
     * 20- APOLICE DE SEGURO
     * 21- MENSALIDADE ESCOLAR
     * 22- PARCELA DE CONSORCIO
     * 23- DIVIDA ATIVA DA UNIAO
     * 24- DIVIDA ATIVA DE ESTADO
     * 25- DIVIDA ATIVA DE MUNICIPIO
     * 31- CARTAO DE CREDITO
     * 32- BOLETO PROPOSTA
     * 99- OUTROS.
     *
     * @var int
     */
    protected $codigoTipoTitulo = 0;

    /*
     * Descrição do tipo de boleto.
     *
     * @var string
     */
    protected $descricaoTipoTitulo;

    /*
     * Código para identificação da autorização de pagamento
     * parcial do boleto. Campo Obrigatório.
     * Domínios: S - SIM N - NÃO
     *
     * @var string
     */
    protected $indicadorPermissaoRecebimentoParcial;

    /*
     * Número de identificação do boleto
     * (correspondente ao SEU NÚMERO), no formato STRING.
     *
     * @var string
     */
    protected $numeroTituloBeneficiario;

    /*
     * Informações adicionais sobre o beneficiário.
     *
     * @var string
     */
    protected $textoCampoUtilizacaoBeneficiario;

    /*
     * Número de identificação do boleto (correspondente ao NOSSO NÚMERO),
     * no formato STRING, com 20 dígitos, que deverá ser formatado da
     * seguinte forma:
     * “000” + (número do convênio com 7 dígitos) + (10 algarismos - se
     * necessário, completar com zeros à esquerda).
     * Campo Obrigatório.
     *
     * @var string
     */
    protected $numeroTituloCliente;

    /*
     * Mensagem definida pelo beneficiário para ser impressa no boleto.
     *
     * @var string
     */
    protected $textoMensagemBloquetoOcorrencia;

    /*
     * Endereço de e-mail para o qual irá o boleto gerado.
     * Campo NÃO OBRIGATÓRIO.
     *
     * @var string
     */
    protected $email;

    /*
     * Quantidade de dias para negativar depois do vencimento do boleto.
     * Valor inteiro >= 0.
     *
     * @var int
     */
    protected $quantidadeDiasNegativacao;

    /*
     * Dados do Avalista
     *
     * @var Object Avalista
     */

    protected $avalista;

    /*
     * Dados do Pagador
     *
     * @var Object Pagador
     */
    protected $pagador;

    protected $multa;

    protected $jurosMora;

    protected $desconto;

    protected $segundoDesconto;

    protected $terceiroDesconto;

    public function __construct($params = [])
    {
        $api = config('diorgesbb.api.homologa');
        if (config('diorgesbb.production')) {
            $api = config('diorgesbb.api.producao');
        }

        foreach ($api as $k => $v) {
            if (method_exists($this, 'set'.ucwords($k))) {
                $this->{'set'.ucwords($k)}($v);
            }
        }

        foreach ($params as $param => $value) {
            $param = str_replace(' ', '', ucwords(str_replace('_', ' ', $param)));
            if (method_exists($this, 'getProtectedFields') && in_array(lcfirst($param), $this->getProtectedFields())) {
                continue;
            }
            if (method_exists($this, 'set'.ucwords($param))) {
                $this->{'set'.ucwords($param)}($value);
            }
        }
    }

    /**
     * @param float $numeroDiasLimiteRecebimento
     */
    public function setNumeroDiasLimiteRecebimento($numeroDiasLimiteRecebimento)
    {
        $this->numeroDiasLimiteRecebimento = $numeroDiasLimiteRecebimento;
    }

    /**
     * @return float
     */
    public function getNumeroDiasLimiteRecebimento()
    {
        return $this->numeroDiasLimiteRecebimento;
    }

    /**
     * @return int
     */
    public function getNumeroConvenio()
    {
        return $this->numeroConvenio;
    }

    /**
     * @param int $numeroConvenio
     */
    public function setNumeroConvenio($numeroConvenio)
    {
        $this->numeroConvenio = $numeroConvenio;
    }

    /**
     * @return int
     */
    public function getNumeroCarteira()
    {
        return $this->numeroCarteira;
    }

    /**
     * @param int $numeroCarteira
     */
    public function setNumeroCarteira($numeroCarteira)
    {
        $this->numeroCarteira = $numeroCarteira;
    }

    /**
     * @return int
     */
    public function getNumeroVariacaoCarteira()
    {
        return $this->numeroVariacaoCarteira;
    }

    /**
     * @param int $numeroVariacaoCarteira
     */
    public function setNumeroVariacaoCarteira($numeroVariacaoCarteira)
    {
        $this->numeroVariacaoCarteira = $numeroVariacaoCarteira;
    }

    /**
     * @return int
     */
    public function getCodigoModalidade()
    {
        return $this->codigoModalidade;
    }

    /**
     * @param int $codigoModalidade
     */
    public function setCodigoModalidade($codigoModalidade)
    {
        $this->codigoModalidade = $codigoModalidade;
    }

    /**
     * @return mixed
     */
    public function getDataEmissao()
    {
        return $this->dataEmissao;
    }

    /**
     * @param mixed $dataEmissao
     */
    public function setDataEmissao($dataEmissao)
    {
        $this->dataEmissao = $dataEmissao;
    }

    /**
     * @return mixed
     */
    public function getDataVencimento()
    {
        return $this->dataVencimento;
    }

    /**
     * @param mixed $dataVencimento
     */
    public function setDataVencimento($dataVencimento)
    {
        $this->dataVencimento = $dataVencimento;
    }

    /**
     * @return int
     */
    public function getValorOriginal()
    {
        return $this->valorOriginal;
    }

    /**
     * @param int $valorOriginal
     */
    public function setValorOriginal($valorOriginal)
    {
        $this->valorOriginal = $valorOriginal;
    }

    /**
     * @return int
     */
    public function getValorAbatimento()
    {
        return $this->valorAbatimento;
    }

    /**
     * @param int $valorAbatimento
     */
    public function setValorAbatimento($valorAbatimento)
    {
        $this->valorAbatimento = $valorAbatimento;
    }

    /**
     * @return int
     */
    public function getQuantidadeDiasProtesto()
    {
        return $this->quantidadeDiasProtesto;
    }

    /**
     * @param int $quantidadeDiasProtesto
     */
    public function setQuantidadeDiasProtesto($quantidadeDiasProtesto)
    {
        $this->quantidadeDiasProtesto = $quantidadeDiasProtesto;
    }

    /**
     * @return mixed
     */
    public function getIndicadorNumeroDiasLimiteRecebimento()
    {
        return $this->indicadorNumeroDiasLimiteRecebimento;
    }

    /**
     * @param mixed $indicadorNumeroDiasLimiteRecebimento
     */
    public function setIndicadorNumeroDiasLimiteRecebimento($indicadorNumeroDiasLimiteRecebimento)
    {
        $this->indicadorNumeroDiasLimiteRecebimento = $indicadorNumeroDiasLimiteRecebimento;
    }

    /**
     * @return mixed
     */
    public function getCodigoAceite()
    {
        return $this->codigoAceite;
    }

    /**
     * @param mixed $codigoAceite
     */
    public function setCodigoAceite($codigoAceite)
    {
        $this->codigoAceite = $codigoAceite;
    }

    /**
     * @return int
     */
    public function getCodigoTipoTitulo()
    {
        return $this->codigoTipoTitulo;
    }

    /**
     * @param int $codigoTipoTitulo
     */
    public function setCodigoTipoTitulo($codigoTipoTitulo)
    {
        $this->codigoTipoTitulo = $codigoTipoTitulo;
    }

    /**
     * @return mixed
     */
    public function getDescricaoTipoTitulo()
    {
        return $this->descricaoTipoTitulo;
    }

    /**
     * @param mixed $descricaoTipoTitulo
     */
    public function setDescricaoTipoTitulo($descricaoTipoTitulo)
    {
        $this->descricaoTipoTitulo = $descricaoTipoTitulo;
    }

    /**
     * @return mixed
     */
    public function getIndicadorPermissaoRecebimentoParcial()
    {
        return $this->indicadorPermissaoRecebimentoParcial;
    }

    /**
     * @param mixed $indicadorPermissaoRecebimentoParcial
     */
    public function setIndicadorPermissaoRecebimentoParcial($indicadorPermissaoRecebimentoParcial)
    {
        $this->indicadorPermissaoRecebimentoParcial = $indicadorPermissaoRecebimentoParcial;
    }

    /**
     * @return mixed
     */
    public function getNumeroTituloBeneficiario()
    {
        return $this->numeroTituloBeneficiario;
    }

    /**
     * @param mixed $numeroTituloBeneficiario
     */
    public function setNumeroTituloBeneficiario($numeroTituloBeneficiario)
    {
        $this->numeroTituloBeneficiario = $numeroTituloBeneficiario;
    }

    /**
     * @return mixed
     */
    public function getTextoCampoUtilizacaoBeneficiario()
    {
        return $this->textoCampoUtilizacaoBeneficiario;
    }

    /**
     * @param mixed $textoCampoUtilizacaoBeneficiario
     */
    public function setTextoCampoUtilizacaoBeneficiario($textoCampoUtilizacaoBeneficiario)
    {
        $this->textoCampoUtilizacaoBeneficiario = $textoCampoUtilizacaoBeneficiario;
    }

    /**
     * @return mixed
     */
    public function getNumeroTituloCliente()
    {
        $convenio = str_pad($this->getNumeroConvenio(), 10, '0', STR_PAD_LEFT);
        $numeroTituloCliente = str_pad($this->numeroTituloCliente, 10, '0', STR_PAD_LEFT);

        return $convenio.$numeroTituloCliente;
    }

    /**
     * @param mixed $numeroTituloCliente
     */
    public function setNumeroTituloCliente($numeroTituloCliente)
    {
        $this->numeroTituloCliente = $numeroTituloCliente;
    }

    /**
     * @return mixed
     */
    public function getTextoMensagemBloquetoOcorrencia()
    {
        return $this->textoMensagemBloquetoOcorrencia;
    }

    /**
     * @param mixed $textoMensagemBloquetoOcorrencia
     */
    public function setTextoMensagemBloquetoOcorrencia($textoMensagemBloquetoOcorrencia)
    {
        $this->textoMensagemBloquetoOcorrencia = $textoMensagemBloquetoOcorrencia;
    }

    /**
     * @return mixed
     */
    public function getEmail()
    {
        return $this->email;
    }

    /**
     * @param mixed $email
     */
    public function setEmail($email)
    {
        $this->email = $email;
    }

    /**
     * @return mixed
     */
    public function getQuantidadeDiasNegativacao()
    {
        return $this->quantidadeDiasNegativacao;
    }

    /**
     * @param mixed $quantidadeDiasNegativacao
     */
    public function setQuantidadeDiasNegativacao($quantidadeDiasNegativacao)
    {
        $this->quantidadeDiasNegativacao = $quantidadeDiasNegativacao;
    }

    /**
     * @return mixed
     */
    public function getPagador()
    {
        return $this->pagador;
    }

    /**
     * @param mixed $pagador
     * @return object Pagador
     */
    public function setPagador($pagador)
    {
        $this->pagador = new Pagador($pagador);

        return $this;
    }

    /**
     * @return mixed
     */
    public function getAvalista()
    {
        return $this->avalista;
    }

    /**
     * @param mixed $avalista
     * @return object Avalista
     */
    public function setAvalista($avalista)
    {
        $this->avalista = new Avalista($avalista);

        return $this;
    }

    /**
     * @return array|mixed
     */
    public function jsonSerialize()
    {
        $arr = [];
        foreach (get_class_methods($this) as $method) {
            if (strpos($method, 'get') !== false) {
                $value = $this->{$method}();

                if (is_numeric($value)) {
                    if ($value > 0) {
                        $arr[lcfirst(str_replace(['set', 'get'], ['', ''], $method))] = $value;
                    }
                } else {
                    if (! empty($value) || strlen(trim($value)) > 0) {
                        $arr[lcfirst(str_replace(['set', 'get'], ['', ''], $method))] = $value;
                    }
                }
            }
        }

        return $arr;
    }
}
